Saga timeout tutorial suggestions (#4589)
Particular/docs.particular.net: tutorials/nservicebus-sagas/2-timeouts/tutorial.md

> For more on the difficulties associated with batch jobs, see [Death to the Batch Job](https://particular.net/blog/death-to-the-batch-job).

> NServiceBus gives us the ability to not only send messages, but also [delay messages](/nservicebus/messaging/delayed-delivery). Delayed delivery gives us the ability to *send messages into the future* which is used by [Saga Timeouts](/nservicebus/sagas/timeouts).

Author: I like it, b ut for some reason I cannot commit your suggestion.

> NServiceBus gives us the ability to not only send messages, but also [delay messages](/nservicebus/messaging/delayed-delivery). Delayed delivery gives us the ability to *send messages into the future* which is used by [Saga Timeouts](/nservicebus/sagas/timeouts).

Reviewer: Ah, good catch. It conflicts with https://github.com/Particular/docs.particular.net/pull/4589/commits/352d8d4ac0f897b64a48746b567f04983b42d9ca

> NServiceBus gives us the ability to not only send messages, but also [delay messages](/nservicebus/messaging/delayed-delivery.md). Delayed delivery gives us the ability to *send messages into the future* which is used by [Saga Timeouts](/nservicebus/sagas/timeouts.md).

Reviewer: ```suggestion
NServiceBus has the ability not only to send and publish messages, but also to [delay messages](/nservicebus/messaging/delayed-delivery.md). Delayed delivery effectively provides the ability to *send messages into the future* which is used by [Saga Timeouts](/nservicebus/sagas/timeouts.md).
```

Reviewer: I re-applied the suggestion @ramonsmits

Author: I like it!

No need to write batch jobs to query every record in the database every night. Instead, each instance is able to manage time in its own workflow, setting virtual alarm clocks to be awakened whenever needed.

> NOTE: Timeouts are guaranteed to be delivered after the passed point in time but might be delayed if the system is very busy.

Reviewer: I'm not sure this reads very well, but I don't have a better proposal right now. I usually explain timeouts as: it's a precise scheduling, it's a not before the expected timeout.

> NOTE: Timeouts are guaranteed to be delivered after the passed point in time but might be delayed if the system is very busy.

Author: ```suggestion
NOTE: Timeouts are guaranteed to not be processed before the given point in time but might be delayed if the system is very busy.
```

Author: @mauroservienti how about this?

Besides the `context`, the `RequestTimeout` method has two parameters of note. One is the `TimeSpan` which tells us how long to wait before sending our trigger message. In this case, it's 20 seconds.

> NOTE: Instead of a `TimeSpan`, we could provide a `DateTime` instance, such as `DateTime.UtcNow.AddDays(10)`. When using this form, make sure if you want to pass a local time which is affected by Daylight Savings Time (DST) changes or to use UTC dates to avoid DST conversion issues.

Reviewer: ```suggestion
NOTE: Instead of a `TimeSpan`, we could provide a `DateTime` instance, such as `DateTime.UtcNow.AddDays(10)`. When using this form, remember that local time is affected by Daylight Savings Time (DST) changes: use UTC dates to avoid DST conversion issues.
```

> NOTE: Instead of a `TimeSpan`, we could provide a `DateTime` instance, such as `DateTime.UtcNow.AddDays(10)`. When using this form, makre sure if you want to pass a local time which is affected by Daylight Savings Time (DST) changes or to use UTC dates to avoid DST conversion issues.

Author: ```suggestion
NOTE: Instead of a `TimeSpan`, we could provide a `DateTime` instance, such as `DateTime.UtcNow.AddDays(10)`. When using this form, make sure if you want to pass a local time which is affected by Daylight Savings Time (DST) changes or to use UTC dates to avoid DST conversion issues.
```
